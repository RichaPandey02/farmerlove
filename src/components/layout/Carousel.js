//Carousel for Home Page......

import React from "react";

import carousl from "../styles/carousl.scss";


const Carousel1 = () => {
  return (
  <div className="ef-carousel">
  <input type="radio" name="unique-name" id="unique-name-1" checked class="ef-carousel__state"/>
  <input type="radio" name="unique-name" id="unique-name-2" className="ef-carousel__state"/>
  <input type="radio" name="unique-name" id="unique-name-3" className="ef-carousel__state"/>
  <div className="ef-carousel__items">
    <img src="https://www.teahub.io/photos/full/80-806201_indian-farmer-hd-wallpaper-download.jpg" alt="" class="ef-carousel__item"/>
    <img src="https://images.pexels.com/photos/6500749/pexels-photo-6500749.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940" alt=" " class="ef-carousel__item" />
    <img src="https://wallpaperaccess.com/full/803482.jpg" alt="" class="ef-carousel__item" />
  </div>
  <div className="ef-carousel__next-group">
    <label for="unique-name-2" className="ef-carousel__next">.</label>
    <label for="unique-name-3" className="ef-carousel__next">.</label>
    <label for="unique-name-1" className="ef-carousel__next">.</label>
  </div>
  <div className="ef-carousel__prev-group">
    <label for="unique-name-3" className="ef-carousel__prev">.</label>
    <label for="unique-name-2" className="ef-carousel__prev">.</label>
    <label for="unique-name-1" className="ef-carousel__prev">.</label>
  </div>
  <div className="ef-carousel__dots">
    <label for="unique-name-1" className="ef-carousel__dot">.</label>
    <label for="unique-name-2" className="ef-carousel__dot">.</label>
    <label for="unique-name-3" className="ef-carousel__dot">.</label>
  </div>
</div>
  );
};

export default Carousel1;

//"https://wallpaperaccess.com/full/4293102.jpg"