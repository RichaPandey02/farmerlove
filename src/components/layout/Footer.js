//Footer page....
import React from "react";
import Footer from "../styles/Footer.css";
const Footerpage = () => {
  return (
    <body className="footer" id="footer">
      <div className="footer__listitem">
        <h5 className="footer__listitem--heading" data-testid="getintouch">
          Get In touch
        </h5>
        <p className="footer__listitem--heading" data-testid="farmersfriend">
          Farmers Friend 👩‍👩‍👧
        </p>
        <p className="footer__listitem--heading" data-testid="Kanpur">
          Kanpur, Uttar Pradesh
        </p>
        <p className="footer__listitem--heading" data-testid="phone">
          8957907453
        </p>
        <p className="footer__listitem--heading" data-testid="email" id="mail">
          richapandey0297@gmail.com
        </p>
      </div>
    </body>
  );
};
export default Footerpage;
