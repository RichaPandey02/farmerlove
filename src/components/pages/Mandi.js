//Mandi page code...
import React from "react";
import { AgGridReact } from "ag-grid-react";
import "ag-grid-community/dist/styles/ag-grid.css";
import "ag-grid-community/dist/styles/ag-theme-alpine.css";
import axios from "axios";
import Aggrid from "../styles/Aggrid.css";
import Card from "./Card";
import Mandii from "../styles/Mandii.css";
import Footer from "../layout/Footer";

const Mandi = () => {
  const rowStyle = {
    background: "transparent",
    color: "#192a56",
  };
  const onGridReady = (params) => {
    console.log("grid is ready");
    fetch("http://localhost:3000/api/mandischema")
      .then((resp) => resp.json())
      .then((resp) => params.api.applyTransaction({ add: resp }));
  };
  const columns = [
    {
      headerName: "STATE",
      field: "STATE",
    },
    {
      headerName: "DIST",
      field: "DIST",
    },
    {
      headerName: "MARKET",
      field: "MARKET",
    },
    {
      headerName: "GODOWN",
      field: "GODOWN",
    },
    {
      headerName: "COLDSTORAGE",
      field: "COLDSTORAGE",
    },
    {
      headerName: "RAILWAY",
      field: "RAILWAY",
    },
    {
      headerName: "ACTION",
      cellRendererFramework: (params) => (
        <div>
          <button onClick={() => cellDeleteing(params)} className="deleteMandi">
            Delete
          </button>
        </div>
      ),
    },
  ];
  const defaultColDef = {
    sortable: true,
    editable: true,
    flex: 1,
    filter: true,
    floatingFilter: true,
    minWidth: 120,
    cellClass: "grid-cell-centered",
  };
  let id;
  const UpadateFunction = (resp) => {
    console.log(resp);
    id = resp.data._id;
    alert("Do you want to update your data?ðŸ™„ðŸ™„");
  };

  const cellEditing = (resp) => {
    id = resp.data._id;

    axios.put(`http://localhost:3000/api/mandi/${id}`, resp.data);
    console.log(resp);
    alert("your data has been updatedðŸ˜ŠðŸ˜Š");
  };
  const cellDeleteing = (resp) => {
    resp.api.applyTransaction({
      remove: [resp.node.data],
    });
    id = resp.data._id;
    axios.delete(`http://localhost:3000/api/mandi/${id}`, resp.data);
    alert("your data has been deleted");
  };

  return (
    <div>
      <Card />
      <br />
      <h1 data-testid="testcase"> Currently Available Mandi's</h1>
      <br />
      <div
        className="ag-theme-alpine manage-table
        container table-responsive-sm"
      >
        <AgGridReact
          columnDefs={columns}
          defaultColDef={defaultColDef}
          onGridReady={onGridReady}
          pagination={true}
          paginationPageSize={10}
          paginationAutoPageSize={true}
          onRowDoubleClicked={UpadateFunction}
          onCellEditingStopped={cellEditing}
          rowStyle={rowStyle}
        />
      </div>
      <br></br>
      <br></br>
      <Footer />
    </div>
  );
};

export default Mandi;
