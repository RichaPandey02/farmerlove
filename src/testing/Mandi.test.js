import Mandi from "../components/pages/Mandi";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Carousel from "../components/layout/Carousel"
import { BrowserRouter } from "react-router-dom";
import Footer from '../components/layout/Footer'
import {configure,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { AgGridReact } from 'ag-grid-react';

import {shallow} from 'enzyme';
configure({adapter:new Adapter()});

// ignore license errors
jest.spyOn(console, 'error').mockImplementation(() => { });
// grid warnings?
jest.spyOn(console, 'warn').mockImplementation(() => { });

const testData = [
  // { make: 'Alfa Romeo', model: 'A', price: 10000 },
  // { make: 'BMW', model: 'B', price: 20000 },
  // { make: 'Citroen', model: 'C', price: 30000 }
];

const setRowData = (wrapper, rowData) => {
  return new Promise(function (resolve, reject) {
    wrapper.setState({ rowData }, () => {
      wrapper.update();
      resolve();
    });
  })
}


const ensureGridApiHasBeenSet = (wrapper) => {
  return new Promise(function (resolve, reject) {
    (function waitForGridReady() {
      if (wrapper.instance().gridApi) {
        resolve(wrapper);
        return;
      }
      setTimeout(waitForGridReady, 100);
    })();
  });
};

describe('Grid Actions Panel', () => {
  let wrapper = null;
  let agGridReact = null;

  beforeEach((done) => {
    wrapper = mount(<Mandi />);
    agGridReact = wrapper.find(AgGridReact).instance();

    ensureGridApiHasBeenSet(wrapper)
      .then(() => setRowData(wrapper, testData))
      .then(() => done());
  });

  afterEach(() => {
    wrapper.unmount();
    wrapper = null;
    agGridReact = null;
  })
  it('renders without crashing', () => {
    expect(wrapper.find('.ag-theme-alpine>.manage-table').exists()).toBeTruthy();
    // expect(wrapper.find('. container table-responsive-sm).exists()).toBeTruthy();
  });
})


describe('shallow Mandi Page',()=>{
    it('render the class name for aggrid',()=>{
        let wrapper=shallow(<Mandi/>)
        console.log(wrapper.debug())
        expect(wrapper.exists('.ag-theme-alpine')).toEqual(true)
    })
  })

afterEach(() => {
  cleanup();
});

it("demo testing", () => {
  expect(true).toBe(true);
});
test("to check whether carausel component rendered", () => {
  render(
    <BrowserRouter>
        <Carousel />
    </BrowserRouter>
  );
});  

test("to check whether Mandi component rendered", () => {
  render(
    <BrowserRouter>
      <Mandi />
    </BrowserRouter>
  );
});

it("the background color should be #192a56", () => {
  const color = "#192a56";
  expect(color).toEqual("#192a56");
});
it("the background should be", () => {
  const background = "transparent";
  expect(background).toEqual("transparent");
});
test("should render the paragraph ", () => {
  render(<Mandi />);
  const element = screen.getByTestId("testcase");
  expect(element).toBeInTheDocument();
  expect(element).not.toContainHTML("<h5>");
});

test("should render the Heading ", () => {
  render(<Mandi />);
  const element = screen.getByTestId("testcase");
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent("Currently Available Mandi's");
});
test("test for testid ", () => {
  render(<Mandi />);
  const element = screen.getByTestId("testcase");
  expect(element).toBeInTheDocument();
  expect(element).not.toContainHTML("<h5>");
});
it("to be equal to 1", () => {
  const num = 1;
  expect(num).toEqual(1);
});

it("should be truthy", () => {
  //Assertions/Expectations
  expect(true).toBeTruthy();
});

test("to check whether Footer component rendered", () => {
  render(
    <BrowserRouter>
        <Footer />
    </BrowserRouter>
  );
});  

it('render the snapshot',()=>{
  let wrapper=shallow(<Mandi/>)
  console.log(wrapper.debug())
  expect(wrapper).toMatchSnapshot();
})
  it('render the class',()=>{
    let wrapper=shallow(<Mandi/>)
    console.log(wrapper.debug())
    // expect(wrapper.exists('.deleteMandi')).toEqual(true)
})

it('render the class name for Aggrid',()=>{
  let wrapper=shallow(<Mandi/>)
  console.log(wrapper.debug())
  expect(wrapper.exists('.ag-theme-alpine')).toEqual(true)
})
it("it should be contain the following text",()=>{
  const result="Currently Available Mandi's"
  expect(result).toBe(result);
})
it("it should be contain the alter ",()=>{
  const result="Do you want to update your data?ðŸ™„ðŸ™„"
  expect(result).toBe(result);
})
it("it should be contain the update text ",()=>{
    
    const result="your data has been updatedðŸ˜ŠðŸ˜Š"
  expect(result).toBe(result);
})
it("it should be contain the delete text ",()=>{
    
  const result="your data has been deleted"
expect(result).toBe(result);
})
