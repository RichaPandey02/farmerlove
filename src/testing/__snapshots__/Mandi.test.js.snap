// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the snapshot 1`] = `
<div>
  <Slider />
  <br />
  <h1
    className="Mandi-heading"
    data-testid="testcase"
  >
     Currently Available Mandi's
  </h1>
  <br />
  <div
    className="ag-theme-alpine manage-table
 container table-responsive-sm"
  >
    <AgGridReact
      columnDefs={
        Array [
          Object {
            "field": "STATE",
            "headerName": "STATE",
          },
          Object {
            "field": "DIST",
            "headerName": "DIST",
          },
          Object {
            "field": "MARKET",
            "headerName": "MARKET",
          },
          Object {
            "field": "GODOWN",
            "headerName": "GODOWN",
          },
          Object {
            "field": "COLDSTORAGE",
            "headerName": "COLDSTORAGE",
          },
          Object {
            "field": "RAILWAY",
            "headerName": "RAILWAY",
          },
          Object {
            "cellRendererFramework": [Function],
            "headerName": "ACTION",
          },
        ]
      }
      defaultColDef={
        Object {
          "cellClass": "grid-cell-centered",
          "editable": true,
          "filter": true,
          "flex": 1,
          "floatingFilter": true,
          "minWidth": 120,
          "sortable": true,
        }
      }
      disableStaticMarkup={false}
      legacyComponentRendering={false}
      maxComponentCreationTimeMs={1000}
      onCellEditingStopped={[Function]}
      onGridReady={[Function]}
      onRowDoubleClicked={[Function]}
      pagination={true}
      paginationAutoPageSize={true}
      paginationPageSize={10}
      rowStyle={
        Object {
          "background": "transparent",
          "color": "#192a56",
        }
      }
    />
  </div>
  <br />
  <br />
  <Footerpage />
</div>
`;
